Runtime/Tracking/Collision/CollisionTracker.cs

```diff
@@ -9,31 +9,61 @@ public class CollisionTracker : CollisionNotifier
     {
         protected virtual void OnCollisionEnter(Collision collision)
         {
+            if ((StatesToProcess & CollisionStates.Enter) == 0)

```

Comment:
Remove all of these in this class, since the superclass already does this in the called `On*` method. This reduces the code duplication.

Reply:
The reasons these are here is because thes `On` methods are part of the unity call and will always happen so we're going to do the event set and call to the inherited method if we don't bomb out early.

Reply:
You just call `OnCollisionStarted` here. `OnCollisionStarted` already does this `if`-check. Therefore you don't need the checks in this subclass.

Reply:
it also calls `eventData.Set(this, false, collision, collision.collider)`

so it will set the event data, which is an extra operation but also is technically also now put the object in the incorrect state because the event data shouldn't be set if the operation shouldn't occur

Reply:
Oh that's what you meant. Yeah especially troublesome as `eventData` is defined on the superclass.